docs/source/development.rst

```diff
@@ -4,63 +4,210 @@
+.. todo::
+   Replace text with label pictures from Github
+
+
+Selecting an issue to work on and assigning people
+--------------------------------------------------
+
+.. note:: If you want to start working on something and don't have a preference,
+   check out the issues labeled "Help wanted"
+
+When you decide to work on an issue, assign yourself to it and describe your
+plans on how you will proceed (in case there is no solution sketch provided in
+the issue description). This way, others may comment on your plans prior to the
+work, which can save a lot of time.
+
+Please make sure that you put all additional information as a comment to the
+issue in case the issue has been discussed elsewhere.
+
+
+Creating a branch
+-----------------
+
+Prior to writing code (or at least before the first commit), you should create a
+branch for solution of the issue. This command creates a new branch called
+``your_branch_name`` and switches your working copy to that branch:
+
+.. code-block:: bash
+
+   $ git checkout -b your_branch_name
+
+
+Writing code
+------------
+
+On the new branch, you can make changes and commit, until your solution is done.
+
+It is worth noting that we are trying to keep our code clean by enforcing some
+code writing rules and guidelines. These are automatically check by Travis CI on
+each push to the Github repository. Here is a list of tools used to check the
+quality of the code:
+
+* `pylint <https://www.pylint.org>`_
+* `pycodestyle <http://pypi.python.org/pypi/pycodestyle>`_
+* `mypy <http://mypy-lang.org>`_
+* `markdownlint <https://github.com/mivok/markdownlint>`_
+
+.. todo:: provide short description to the tools, check that markdownlint has
+          correct URL
+
+You can run the tests on your local machine by using scripts (and requirements)
+from the ``tests/`` directory of this package,
+
+This is a usual mantra that you can use for committing and pushing to the remote
+branch in the repository:
+
+.. code-block:: bash
+
+   $ git add .
+   $ git commit -m 'your commit message'
+   $ git push origin your_branch_name
+
+.. note:: If you are working on a branch with someone else, it is always a good
+          idea to do a ``git pull --rebase`` before pushing. This command
+          updates your branch with remote changes and apply your new commits on
+          top of them.
+
+.. warning:: If your commit message contains the string ``[ci skip]`` the
+	     continuous integration tests are not run. However, try not to use
+	     this feature unless you know what you're doing.
+
+
+Creating a pull request
+-----------------------
+
+Whenever you want to add a feature or push a bugfix, you should make a new pull
+request, which can be reviewed and merged by someone else. The typical workflow
+should be as follows:
+
+1. Create a new branch, make your changes and push them to the repository.
+
+2. You should now see the new branch on the Github project page. When you open
```

Comment:
Mohlo by se rozlišit, zda přispěvatel má push access (pro takové ale asi tento dokument primárně není), nebo si musel udělat fork.
A propos: nebylo by lepší odkázat se třeba na https://guides.github.com/introduction/flow/ a pak jen popsat, co je v NM navíc či jinak?